### PR TITLE
Use Seed CRD for the conformance-tester

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Makes the conformance tester use the seed crd instead of the `datacenters.yaml` file

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
none
```

/assign @kdomanski

### CODE REVIEW

**Comment by kdomanski** [comment]
`cmd/conformance-tests/main.go:134:1: cyclomatic complexity 35 of func `main` is high (> 30) (gocyclo)`

You could extract everything related to manipulating `opts` into a separate function.

File: .golangci.yml [hunk 1]
```diff
@@ -23,9 +23,7 @@ linters:
 
 issues:
   exclude:
-  - cyclomatic complexity 32 of func `main` is high
-  - cyclomatic complexity 33 of func `main` is high
-  - cyclomatic complexity 43 of func `main` is high
+  - cyclomatic complexity [0-9]+ of func `main` is high
```
Comment (kdomanski) [inline .golangci.yml:26]: The way to cure a fever is not to throw out the thermometer. And the sheer audacity of calling the commit "fix" is mind-boggling.

**Comment by kdomanski** [comment]
cc @mrIncompetent

File: .golangci.yml (hunk 1 above)
Comment (alvaroaleman) [reply .golangci.yml:26]: Its not worse than having five exclude rules  with different numbers that will gladly match any `main` func

Comment (kdomanski) [reply .golangci.yml:26]: An proper alternative would be to split the over-sized function into smaller ones for better readability.

Comment (mrIncompetent) [reply .golangci.yml:26]: We had that discussion on Slack at some point.
I think the main outcome was:
- Ignoring the linter in mains is acceptable
- Some dislike the linter, some like it

Comment (alvaroaleman) [reply .golangci.yml:26]: I will not write shitty factored code just to satisfy a linter. A func that does a lot of top-level conditional logic without nesting it is a lot easier to read than factoring that stuff out into a distinct func that to call that distinct func once

**Pull request merged**